tools/code_format/check_format.py

```diff
@@ -372,40 +260,44 @@ def check_namespace(self, file_path):
     # To avoid breaking the Lyft import, we just check for path inclusion here.
     def allow_listed_for_protobuf_deps(self, file_path):
         return (
-            file_path.endswith(PROTO_SUFFIX) or file_path.endswith(REPOSITORIES_BZL)
-            or any(path_segment in file_path for path_segment in GOOGLE_PROTOBUF_ALLOWLIST))
+            file_path.endswith(self.config.suffixes["proto"])
+            or file_path.endswith(self.config.suffixes["repositories_bzl"])
+            or any(file_path.startswith(path) for path in self.config.paths["protobuf"]["include"]))
 
     # Real-world time sources should not be instantiated in the source, except for a few
     # specific cases. They should be passed down from where they are instantied to where
     # they need to be used, e.g. through the ServerInstance, Dispatcher, or ClusterManager.
     def allow_listed_for_realtime(self, file_path):
         if file_path.endswith(".md"):
             return True
-        return file_path in REAL_TIME_ALLOWLIST
+        return file_path in self.config.paths["real_time"]["include"]
 
     def allow_listed_for_register_factory(self, file_path):
         if not file_path.startswith("./test/"):
             return True
 
-        return any(file_path.startswith(prefix) for prefix in REGISTER_FACTORY_TEST_ALLOWLIST)
+        return any(
+            file_path.startswith(prefix)
+            for prefix in self.config.paths["register_factory_test"]["include"])
 
     def allow_listed_for_serialize_as_string(self, file_path):
-        return file_path in SERIALIZE_AS_STRING_ALLOWLIST
+        return file_path in self.config.paths["serialize_as_string"]["include"]
 
     def allow_listed_for_std_string_view(self, file_path):
-        return file_path in STD_STRING_VIEW_ALLOWLIST
+        return file_path in self.config.paths["std_string_view"]["include"]
 
     def allow_listed_for_json_string_to_message(self, file_path):
-        return file_path in JSON_STRING_TO_MESSAGE_ALLOWLIST
+        return file_path in self.config.paths["json_string_to_message"]["include"]
 
     def allow_listed_for_histogram_si_suffix(self, name):
-        return name in HISTOGRAM_WITH_SI_SUFFIX_ALLOWLIST
+        return name in self.config.suffixes["histogram_with_si"]["include"]
 
     def allow_listed_for_std_regex(self, file_path):
-        return file_path.startswith("./test") or file_path in STD_REGEX_ALLOWLIST
+        return file_path.startswith(
+            "./test") or file_path in self.config.paths["std_regex"]["include"]
 
     def allow_listed_for_grpc_init(self, file_path):
-        return file_path in GRPC_INIT_ALLOWLIST
+        return file_path in self.config.paths["grpc_init"]["include"]
```

Comment:
Why not make this a proto? That way we can still do YAML -> JSON -> proto3, but we end up with a typed schema, instead of this dictionary stuff. It would strictly be more readable and scalable.

Reply:
proto is great for wire/message/serialization formats - none of that is relevant here

i think the point is moot anyhow because as it stands this _cannot_ add any deps

when this gets moved to bazel (and probs pytooling) we can add deps - i still see no reason even then to add protobuf - its a lot of overhead for no real benefit

in this PR i added type hinting - mostly for my/developers benefit - its not linted/checked anywhere. In pytooling i have added some code (based on upstream `typecast`) that does ensure the parsed config files are valid/well-formed

what this PR does do is take some of the worst code in codebase (readability/usability/performance - and up until recently code errors) and factors out a big chunk of it into a format that end users/devs can actually work with in a reasonable fashion